src/models/Deck.ts

```diff
@@ -110,7 +110,16 @@ export class Deck {
     ids?.map((v, i) => (cards[i].noteId = v));
   }
 
-  async pushNewCardsToAnki() {
+  // Calls anki to update the fields of all the passed cards.
+  private async _pushUpdatedCardsToAnki(cards: Card[]) {
+    cards?.map((card) => this.ankiService?.updateFields(card));
```

Comment:
This method is async btu there's no `await` being used.
You're also not returning any value so `map` is redundant here, you may instead just want to use forEach.

Do you want to wait for this function to finish? If so you could return the promises into a Promise.all() and await that

Reply:
I'm going to drop the `async` on this function as I don't care exactly when this finishes.
I'll also switch to `forEach`.
I'll remove the optional chaining you mentioned.

Reply:
I think we do need to care when operations like this finish though. Due to the progress bar that’s shown in VSCode, it needs to know when to stop showing.

Reply:
I've taken your advice and ensured I'm `await`ing the async `_pushUpdatedCardsToAnki` function now.